src/components/select/bl-select.ts

```diff
@@ -453,6 +578,44 @@ export default class BlSelect<ValueType extends FormValue = string> extends Form
     );
   }
 
+  private _handleSearchEvent() {
+    this._onBlSearch(this._searchText);
+  }
+
+  private _handleSearchOptions(e: InputEvent): void {
+    if (!this.searchBar) return;
+
+    this._searchText = (e.target as HTMLInputElement).value;
+
+    this._handleSearchEvent();
+
+    this._connectedOptions.forEach(option => {
+      const isVisible = option.textContent?.toLowerCase().includes(this._searchText.toLowerCase());
+
+      option.hidden = !isVisible;
+    });
+
+    this._selectedOptions = this.options.filter(option => option.selected);
+
+    this._handleLastVisibleSearchedOption();
+
+    this.requestUpdate();
```

Comment:
It is kinda scary but necessary. Let's wait for the feedbacks for the next week till the stable.